clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/MigrateLoadBalancerStrategy.java

```diff
@@ -0,0 +1,452 @@
+
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.*;
+import com.amazonaws.services.elasticloadbalancing.AmazonElasticLoadBalancing;
+import com.amazonaws.services.elasticloadbalancing.model.*;
+import com.netflix.spinnaker.clouddriver.aws.AwsConfiguration.DeployDefaults;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.loadbalancer.LoadBalancerMigrator.LoadBalancerLocation;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.loadbalancer.MigrateLoadBalancerResult;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.MigrateSecurityGroupReference;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.MigrateSecurityGroupResult;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.SecurityGroupLookupFactory.SecurityGroupLookup;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.SecurityGroupLookupFactory.SecurityGroupUpdater;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.SecurityGroupMigrator;
+import com.netflix.spinnaker.clouddriver.aws.deploy.ops.securitygroup.SecurityGroupMigrator.SecurityGroupLocation;
+import com.netflix.spinnaker.clouddriver.aws.model.SubnetTarget;
+import com.netflix.spinnaker.clouddriver.aws.provider.view.AmazonVpcProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.aws.services.RegionScopedProviderFactory;
+import com.netflix.spinnaker.clouddriver.helpers.OperationPoller;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class MigrateLoadBalancerStrategy {
+
+  protected SecurityGroupLookup sourceLookup;
+  protected SecurityGroupLookup targetLookup;
+
+  abstract AmazonClientProvider getAmazonClientProvider();
+
+  abstract RegionScopedProviderFactory getRegionScopedProviderFactory();
+
+  abstract MigrateSecurityGroupStrategy getMigrateSecurityGroupStrategy();
+
+  abstract DeployDefaults getDeployDefaults();
+
+  /**
+   * Generates a result set describing the actions required to migrate the source load balancer to the target.
+   *
+   * @param sourceLookup    a security group lookup cache for the source region
+   * @param targetLookup    a security group lookup cache for the target region (may be the same object as the sourceLookup)
+   * @param source          the source load balancer
+   * @param target          the target location
+   * @param subnetType      the subnetType in which to migrate the load balancer (should be null for EC Classic migrations)
+   * @param applicationName the name of the source application
+   * @param dryRun          whether to actually perform the migration
+   * @return the result set
+   */
+  public MigrateLoadBalancerResult generateResults(SecurityGroupLookup sourceLookup, SecurityGroupLookup targetLookup,
```

Comment:
It would probably be worth making this method (and the equivalent on MigrateSecurityGroupStrategy) `synchronized` to make it clear that calling this changes the state of the class